vassal-app/src/main/java/VASSAL/tools/DataArchive.java

```diff
@@ -169,6 +170,26 @@ public InputStream getImageInputStream(String fileName)
       "'" + imageDir + fileName + "' not found in " + getName());
   }
 
+  /**
+   * Return the size in bytes of a file in the Archive
+   * @param fileName File name (include path if required)
+   * @return size in bytes
+   */
+  public long getFileSize(String fileName) {
```

Comment:
Is there a reason you're doing this instead of using `FileArchive.getSize()`?

Reply:
Just unfamiliarity with the code. Will FileArchive,getSize() return the compressed or uncompressed size? Is there an easy way to get the compressed size? That would be better as it relates directly to the issue of reducing the module size.

Reply:
It's uncompressed size---but most images barely compress at all, so uncompressed size is a pretty good proxy for compressed size.

Reply:
Turned out to be easy to access the compressed size. Try this new version